tests/Operation/WatchFunctionalTest.php

```diff
@@ -900,12 +897,23 @@ public function testMaxAwaitTimeMS()
         $this->insertDocument(['_id' => 1]);
 
         /* Advancing the change stream again will issue a getMore, but the
-         * server should not block since a document has been inserted. */
-        $startTime = microtime(true);
-        $changeStream->next();
-        $duration = microtime(true) - $startTime;
-        $this->assertLessThan($pivot, $duration);
+         * server should not block since a document has been inserted.
+         * For sharded clusters, we have to repeat the getMore iteration until
+         * the cursor is valid since the first getMore commands after an insert
+         * may not return any data. Only the time of the last getMore command is
+         * taken. */
+        for ($i = 0; $i < 10; $i++) {
```

Comment:
Any reason to use 10 attempts here vs. the five that you used for `advanceCursorUntilValid()`?

On a separate note, I think a more conservative change would be something like:

```
$attempts = $this->isShardedCluster() ? 5 : 1;
```

And then use `$attempts` in the `for` loop. That would ensure replica sets only ever make one attempt. Granted, I don't expect that a replica set would ever execute multiple `getMore` commands unless we broke this test in some other way.